src/test/HostActivationTests/MultilevelSDKLookup.cs

```diff
@@ -35,36 +34,29 @@ public class MultilevelSDKLookup : IDisposable
         private readonly string _regSdkBaseDir;
         private readonly string _exeSelectedMessage;
         private readonly string _regSelectedMessage;
-        private readonly string _sdkDir;
         private readonly string _multilevelDir;
 
         private const string _dotnetSdkDllMessageTerminator = "dotnet.dll]";
 
         public MultilevelSDKLookup()
         {
-            // From the artifacts dir, it's possible to find where the sharedFrameworkPublish folder is. We need
-            // to locate it because we'll copy its contents into other folders
-            string artifactsDir = Environment.GetEnvironmentVariable("TEST_ARTIFACTS");
-            string builtDotnet = Path.Combine(artifactsDir, "sharedFrameworkPublish");
-
             // The dotnetMultilevelSDKLookup dir will contain some folders and files that will be
             // necessary to perform the tests
-            string baseMultilevelDir = Path.Combine(artifactsDir, "dotnetMultilevelSDKLookup");
+            string baseMultilevelDir = Path.Combine(TestArtifact.TestArtifactsPath, "dotnetMultilevelSDKLookup");
             _multilevelDir = SharedFramework.CalculateUniqueTestDirectory(baseMultilevelDir);
 
-            // The three tested locations will be the cwd, the user folder and the exe dir. cwd and user are no longer supported.
+            // The tested locations will be the cwd, user folder, exe dir, and registered directory. cwd and user are no longer supported.
             //     All dirs will be placed inside the multilevel folder
-
             _currentWorkingDir = Path.Combine(_multilevelDir, "cwd");
             _userDir = Path.Combine(_multilevelDir, "user");
             _exeDir = Path.Combine(_multilevelDir, "exe");
             _regDir = Path.Combine(_multilevelDir, "reg");
 
-            // It's necessary to copy the entire publish folder to the exe dir because
-            // we'll need to build from it. The CopyDirectory method automatically creates the dest dir
-            SharedFramework.CopyDirectory(builtDotnet, _exeDir);
+            DotNet = new DotNetBuilder(_multilevelDir, Path.Combine(TestArtifact.TestArtifactsPath, "sharedFrameworkPublish"), "exe")
```

Comment:
The constructor runs for every single test case... do we need to repeat all of it every time?

Feel free to ignore it for this change and maybe we can get back to this later...

Reply:
The way the tests were written, unfortunately yes. Each test adds SDKs to the different directories and checks what is found / used (then adds more, checks again) - with the assumption that the initial state is the clean state (as is set up by the constructor) and without removing what was added by the test. I'm going to leave it for this change, but I'd agree we can revisit how these are structured.

Reply:
We could do something similar to the `DotNetCliCustomizer` https://github.com/dotnet/core-setup/blob/6a21cad8eec0c1a63e8390c5f437d1f2d4872f1f/src/test/HostActivationTests/FrameworkResolution/DotNetCliExtensions.cs#L25
It exposes methods to customize the folder and then reverts to original state by tracking all the changes along the way.
But I agree this can be done later on.